### PR TITLE
Fix release workflow

### PR DESCRIPTION
**Pull request by unguiculus**
* Install required tooling
* Create Git tag locally instead of via GitHub API so Goreleaser
  can find it

Signed-off-by: Reinhard Nägele <unguiculus@gmail.com>

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -16,11 +16,36 @@ jobs:
...
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+          tag='${{ github.event.inputs.version }}'
+          git tag --annotate --message "Tag for release $tag" "$tag"
+          git push "https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY" "refs/tags/$tag"
+
+      - name: Check License Headers
+        run: mage -v checkLicenseHeaders
+
+      - name: Lint
```
Comment (davidkarlsen) [inline .github/workflows/release.yaml:47]: should maybe the checks happen before the tag?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -16,11 +16,36 @@ jobs:
...
+        run: ./setup.sh
+
+      - name: Tag
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+          tag='${{ github.event.inputs.version }}'
+          git tag --annotate --message "Tag for release $tag" "$tag"
+          git push "https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY" "refs/tags/$tag"
```
Comment (davidkarlsen) [inline .github/workflows/release.yaml:42]: afaik the checkout action configures git so you can push back to the same repo w/o having to deal with tokens?

Comment (unguiculus) [reply .github/workflows/release.yaml:42]: will check that, thanks

File: .github/workflows/release.yaml (hunk 1 above)
Comment (unguiculus) [reply .github/workflows/release.yaml:47]: Ouch, good catch

File: .github/workflows/release.yaml (hunk 2 above)
Comment (davidkarlsen) [reply .github/workflows/release.yaml:42]: there is however a need to configure author etc: https://github.com/actions/checkout/issues/233 - but I found an action which can do that. give me 2 sec

Comment (unguiculus) [reply .github/workflows/release.yaml:42]: I'm already doing that.

Comment (davidkarlsen) [reply .github/workflows/release.yaml:42]: https://github.com/fregante/setup-git-token - though marginally helpful.

Comment (unguiculus) [reply .github/workflows/release.yaml]: Verify pushing works without specifying the token

File: .github/workflows/release.yaml (hunk 1 above)
Comment (unguiculus) [reply .github/workflows/release.yaml:39]: Fixed

**Review by davidkarlsen — APPROVED** [review]

**Comment by unguiculus** [comment]
> If all goes smoothly (no unanticipated gotchas etc) we may want to roll out something similar to other projects

Definitely. My plan was to do this next for chart-testing. Or, if anyone else wants to pick that up, feel free.

**Pull request merged**